extensions/arc/deployment/src/test/java/io/quarkus/arc/test/synthetic/restrictTypes/SyntheticBeanBuildItemRestrictTypesTest.java

```diff
@@ -0,0 +1,109 @@
+package io.quarkus.arc.test.synthetic.restrictTypes;
+
+import java.util.function.Consumer;
+
+import jakarta.enterprise.inject.Vetoed;
+
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.SyntheticCreationalContext;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.arc.processor.BuiltinScope;
+import io.quarkus.builder.BuildChainBuilder;
+import io.quarkus.builder.BuildContext;
+import io.quarkus.builder.BuildStep;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class SyntheticBeanBuildItemRestrictTypesTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(SyntheticBeanBuildItemRestrictTypesTest.class, FooCreator.class, FooInterface.class, Foo.class,
+                            FooSubclass.class, Charlie.class, CharlieSubclass.class, CharlieInterface.class, BarInterface.class,
+                            BazInterface.class))
+            .addBuildChainCustomizer(buildCustomizer());
+
+    static Consumer<BuildChainBuilder> buildCustomizer() {
+        return new Consumer<BuildChainBuilder>() {
+
+            @Override
+            public void accept(BuildChainBuilder builder) {
+                builder.addBuildStep(new BuildStep() {
+
+                    @Override
+                    public void execute(BuildContext context) {
+                        context.produce(SyntheticBeanBuildItem.create(FooSubclass.class)
+                                .addTypeClosure(FooSubclass.class)
+                                .removeTypes(DotName.createSimple(CharlieSubclass.class))
+                                .removeTypes(DotName.createSimple(FooSubclass.class))
+                                .removeTypes(Type.create(BazInterface.class))
+                                .scope(BuiltinScope.SINGLETON.getInfo())
+                                .unremovable()
+                                .creator(FooCreator.class)
+                                .done());
+                    }
+                }).produces(SyntheticBeanBuildItem.class).build();
+            }
+        };
+    }
+
+    @Test
+    public void testRestrictingBeanTypes() {
```

Comment:
The class name and the test method name still refere to "restricting" types, which is probably not intentional?

Reply:
Good point, so does the package name - will change :+1: